pennylane/workflow/qnode.py

```diff
@@ -915,10 +916,19 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         if old_interface == "auto":
             self.interface = qml.math.get_interface(*args, *list(kwargs.values()))
 
+        # Before constructing the tape, we pass the device to the
+        # debugger to ensure they are compatible if there are any
+        # breakpoints in the circuit
+        if PLDB.is_active_dev():
+            PLDB.reset_active_dev()
+
+        PLDB.add_device(self.device)

```

Comment:
Just a quick check, but this setup shouldn't have any problem with concurrent execution, right? If we in parallel create 2 qnodes with 2 devices, what happens here if they try to modify the class at the same time?

Reply:
Is there a way to implement this without a queuing manager approach?
If not, as a quick validation, can we try using the debugger with `default.qubit` and max_workers >= 2 to see what happens.